lib/gem_content/package.rb

```diff
@@ -0,0 +1,31 @@
+# A wrapper around Gem::Package to add behaviors we need for working with
+# individual files in memory.
+class GemContent::Package
+  attr_reader :package
+
+  def initialize(gem)
+    @package = gem.is_a?(::Gem::Package) ? gem : ::Gem::Package.new(gem)
+  end
+
+  delegate :verify, :spec, :contents, to: :package
+
+  def each(&)
+    open_data_tar do |data_tar|
+      data_tar.each do |entry|
+        begin
+          yield GemContent::Entry.from_package_entry(entry)
+        rescue StandardError => e
+          Rails.error.report(e, handled: true)
```

Comment:
can we report the gem full name and entry data as context in the call to `Rails.error.report` ?